### PR TITLE
add article votes functionality - ticket 7

### PR DESCRIPTION
**Pull request by veramferreira**

### CODE REVIEW

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -35,6 +38,30 @@ export default function ArticleCard() {
     );
   }
 
+  function handleUpVote() {
+    setVoted((currVotes) => {
```
Comment (Augs0) [inline src/components/ArticleCard.jsx]: Your voting logic seems to only optimistically render and never patches to the database. It looks like you've got a bit confused here about how to handle the vote. We need to:

- optimistically render using the state for a newVote here. We can use our setVoted function to add one to the previous state - no need to return an object
- then make a call to the database with patch to update the actual votes

File: src/components/ArticleCard.jsx [hunk 2]
```diff
@@ -43,13 +70,27 @@ export default function ArticleCard() {
...
           src={article_img_url}
           alt={`something related to ${title}`}
         />
-        <p className="article--body">{body}</p>
-        <p className="article--author">written by: {author}</p>
+        <div className="article--body">
+          <p>{body}</p>
+          <p className="bold">Did you like this article?</p>
+          <div className="article--votes">
+            <InsertEmoticonOutlinedIcon
```
Comment (Augs0) [inline src/components/ArticleCard.jsx]: For accessibility reasons, we would need to let users know what the emojis represent. This can be done with an aria-label, for example. There might be something readily available through mui to add a property to the inserted component

**Review by Augs0 — CHANGES_REQUESTED** [review]
Hey Vera, your app looks super nice but unfortunately we haven't quite got the logic we need for voting yet. Please see my comments on that file to fix the error

File: src/components/ArticleCard.jsx (hunk 1 above)
Comment (veramferreira) [reply src/components/ArticleCard.jsx]: Yeah, I realised this as soon as I submitted a PR. 🤦‍♀️ sorry! I've been working on it for the last 45 minutes, but I might need to put a /nchelp :/

File: src/components/ArticleCard.jsx (hunk 2 above)
Comment (veramferreira) [reply src/components/ArticleCard.jsx]: Ok! I tried an alt attribute, but didn't work 😅 I'll investigate the aria-labels.

File: src/components/ArticleCard.jsx [hunk 3]
```diff
@@ -35,6 +39,38 @@ export default function ArticleCard() {
...
+        });
+        setVoteMessage("Upvoted! Nice!");
+      })
+      .catch((err) => {
+        console.log(err);
+        setVoteMessage("Oooops! Something went wrong. Try again later.");
+      });
+  }
+
+  function handleDownVote() {
```
Comment (JVal-NC) [inline src/components/ArticleCard.jsx:60]: Vera, apologies, my fault but this currently isnt optimistic. We want to update the local state then do the API request.

Inside the catch block, as this will indicate the request has failed, we will want to undo the change we did before the API call.

This is why it is optimistic, we're optimistic that the request is going to run without errors before executing it.

File: src/components/ArticleCard.jsx [hunk 4]
```diff
@@ -35,6 +39,38 @@ export default function ArticleCard() {
...
+      .then(() => {
+        setVoted(newVote);
+        setArticle((currArticle) => {
+          return { ...currArticle, votes: currArticle.votes + 1 };
+        });
+        setVoteMessage("Upvoted! Nice!");
+      })
+      .catch((err) => {
+        console.log(err);
+        setVoteMessage("Oooops! Something went wrong. Try again later.");
```
Comment (JVal-NC) [inline src/components/ArticleCard.jsx]: These error messages dont seem to be coming through? If we use inspect and then set chrome toe offline mode they dont show, will drop in to explain this.

**Review by JVal-NC — COMMENTED** [review]
Vera good work so far, a few comments included to consider but I will drop in to your room to have a chat.

**Review by xViolaine — APPROVED** [review]
Really nice implementation of the feedback given. Good job :)
